met-web/src/components/common/Modals/FeedbackModal.tsx

```diff
@@ -0,0 +1,342 @@
+import CloseIcon from '@mui/icons-material/Close';
+import { ReactComponent as CheckIcon } from 'assets/images/check.svg';
+import { ReactComponent as ExclamationIcon } from 'assets/images/exclamation.svg';
+import { ReactComponent as LightbulbIcon } from 'assets/images/lightbulb.svg';
+import { ReactComponent as ThinkingIcon } from 'assets/images/thinking.svg';
+import { ReactComponent as VeryDissatisfiedIcon } from 'assets/images/emojiVeryDissatisfied.svg';
+import { ReactComponent as DissatisfiedIcon } from 'assets/images/emojiDissatisfied.svg';
+import { ReactComponent as NeutralIcon } from 'assets/images/emojiNeutral.svg';
+import { ReactComponent as SatisfiedIcon } from 'assets/images/emojiSatisfied.svg';
+import { ReactComponent as VerySatisfiedIcon } from 'assets/images/emojiVerySatisfied.svg';
+import { useState } from 'react';
+import { ConditionalComponent, MetBody, MetHeader3, MetHeader4, modalStyle, PrimaryButton } from '..';
+import { BaseTheme } from 'styles/Theme';
+import { createDefaultFeedback } from 'models/feedback';
+
+export const FeedbackModal = () => {
+    const [isSubmitted, setIsSubmitted] = useState(false);
+    const [isOpen, setIsOpen] = useState(false);
+    const [feedbackFormData, setFeedbackFormData] = useState(createDefaultFeedback());
+    const { comment, rating, commentType } = feedbackFormData;
+
+    const StyledRating = styled(Rating)(({ theme }) => ({
+        '& .MuiRating-iconEmpty .MuiSvgIcon-root': {
+            opacity: '0.5',
+        },
+        textAlign: 'center',
+    }));
+
+    const customIcons: {
+        [index: string]: {
+            icon: React.ReactElement;
+            label: string;
+        };
+    } = {
+        5: {
+            icon: <SvgIcon fontSize="large" component={VeryDissatisfiedIcon} viewBox="0 0 64 64" sx={{ marginX: 1 }} />,
+            label: 'Very Dissatisfied',
+        },
+        4: {
+            icon: <SvgIcon fontSize="large" component={DissatisfiedIcon} viewBox="0 0 64 64" sx={{ marginX: 2 }} />,
+            label: 'Dissatisfied',
+        },
+        3: {
+            icon: <SvgIcon fontSize="large" component={NeutralIcon} viewBox="0 0 64 64" sx={{ marginX: 1 }} />,
+            label: 'Neutral',
+        },
+        2: {
+            icon: <SvgIcon fontSize="large" component={SatisfiedIcon} viewBox="0 0 64 64" sx={{ marginX: 2 }} />,
+            label: 'Satisfied',
+        },
+        1: {
+            icon: <SvgIcon fontSize="large" component={VerySatisfiedIcon} viewBox="0 0 64 64" sx={{ marginX: 1 }} />,
+            label: 'Very Satisfied',
+        },
+    };
+
+    function IconContainer(props: IconContainerProps) {
+        const { value, ...other } = props;
+        return <span {...other}>{customIcons[value].icon}</span>;
+    }
+
+    function getLabelText() {
+        if (rating == 0) return;
+        return `${customIcons[rating].label}`;
+    }
+
+    function getPlaceholderText() {
```

Comment:
Having set the exact type of the commentType, do you think we can change this to a simpler object? like customIcons

Reply:
Simplified